README.md

```diff
@@ -1,289 +1,607 @@
-
-**Note** - a few updates / enhancements are being proposed to the current `ServiceBinding` CR, tracked [here](https://github.com/application-stacks/service-binding-specification/issues/16#issuecomment-605309629).
-
-Sample CR
+## Well-known Secret Entries
+
+Other than the required `kind` entry and the recommended `provider` entry, there are no other reserved `Secret` entries.  In the interests of consistency, if a `Secret` includes any of the following entry names, the entry value **MUST** meet the specified requirements:
+
+| Name | Requirements
+| ---- | ------------
+| `host` | A DNS-resolvable host name or IP address
+| `port` | A valid port number
+| `uri` | A valid URI as defined by [RFC3986](https://tools.ietf.org/html/rfc3986)
+| `username` | A string-based username credential
+| `password` | A string-based password credential
+| `certificates` | A collection of PEM-encoded X.509 certificates, representing a certificate chain used in mTLS client authentication
+| `privateKey` | A PEM-encoded private key used in mTLS client authentication
+
+`Secret` entries that do not meet these requirements **MUST** use different entry names.
+
+## Example Secret
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: production-db
+stringData:
+  kind: mysql
+  provider: bitnami
+  host: localhost
+  port: 3306
+  username: root
+  password: root
 ```
+
+# Application Projection
+
+A Binding `Secret` **MUST** be volume mounted into a container at in `$SERVICE_BINDINGS_ROOT/<binding-name>` with directory names matching the name of the binding.  Binding names **MUST** match `[a-z0-9\-\.]{1,253}`.  The `$SERVICE_BINDINGS_ROOT` environment variable **MUST** be declared and can point to any valid file system location.
+
+The `Secret` **MUST** contain a `kind` entry with a value that identifies the abstract classification of the binding.  It is **RECOMMENDED** that the `Secret` also contain a `provider` entry with a value that identifies the provider of the binding.  The `Secret` **MAY** contain any other entry.
+
+The contents of a secret entry may be anything representable as bytes on the file system including, but not limited to, a literal string value (e.g. `db-password`), a language-specific binary (e.g. a Java `KeyStore` with a private key and X.509 certificate), or an indirect pointer to another system for value resolution (e.g. `vault://production-database/password`).
+
+The collection of files within the directory **MAY** change between container launches.  The collection of files within the directory **SHOULD NOT** change during the lifetime of the container.
+
+## Example Directory Structure
+
+```plain
+$SERVICE_BINDING_ROOT
+├── account-database
+│   ├── kind
+│   ├── connection-count
```

Comment:
should we use `provider` instead of `connection-count` in our examples, for consistency?

Reply:
Since `provider` was mentioned in the spec, I was trying to illustrate a non-secret secret, and the optionality of `provider`.  I don't feel strongly, so make a proposal of an alternative and I'll probably take it.